client/lib/performance-tracking/index.js

```diff
@@ -0,0 +1,22 @@
+/**
+ * Internal dependencies
+ */
+import config from 'config';
+
+/**
+ * External dependencies
+ */
+import { start } from '@automattic/browser-data-collector';
+
+export const trackNavigationStart = ( pageName ) => {
+	return ( context, next ) => {
+		if ( config.isEnabled( 'rum-tracking/logstash' ) ) {
+			start( pageName || 'calypso', {
+				fullPageLoad: context.init || false,
+			} );
+		}
+		next();
+	};
+};
```

Comment:
To help tree shaking, I like the convention where the `index.js` file reexports bindings from submodules and contains nothing else. This module defines one function inline, and another is reexported. That feels a bit inconsistent.